contents/blog/improving-posthog-deployments.md

```diff
@@ -0,0 +1,127 @@
+---
+date: 2022-03-22
+title: "Improving PostHog deployments"
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+featuredImage: ../images/blog/simpler-self-deployments.png
+featuredImageType: full
+categories: ["Engineering", "Product updates", "Launch week"]
+author: ["harry-waye", "guido-iaquinti"]
+---
+
+When PostHog was born in 2020, it was a simple Python application (Django + Celery) backed by a PostgreSQL datastore. Troubleshooting was easy, while the low barrier of entry meant fast adoption and more feedback on where to take the product.
+
+This was great at the start, but it couldn't last. More customers meant more data to ingest and analyze. Our ingestion couldn’t keep up with the volumes of events we were receiving, and our event queries using PostgreSQL were slow. If we wanted PostHog to scale, we had to revisit our stack:
+
+Introducing Kafka helped us to decouple our data streams and our processing pipeline, while [ClickHouse helped us to deliver very fast queries](/blog/clickhouse-announcement) on a large volume of data and more horizontal scaling capabilities. These additions helped us scale, but they came at a cost: complexity.
+
+Here's what our architecture looks like now.
+
+```mermaid
+graph TD
+    CLIENT[Client Library]
+    DECIDE["/decide API"]
+    CAPTURE[Capture API]
+    PLUGINS[Plugin server]
+    PERSONS["PostgreSQL (persons table)"]
+    Kafka2[Kafka]
+
+    CLIENT -..-> DECIDE
+
+    CLIENT -..-> CAPTURE
+    CAPTURE --> Kafka
+
+    Kafka <-..- PLUGINS
+
+    PLUGINS <--> PERSONS
+    PLUGINS --> Kafka2
+
+    Kafka2 <-..- ClickHouse
+```
+
+While powerful, this new architecture has created the following challenges:
+
+* We now have additional distributed services like Kafka, ClickHouse and Zookeeper to deploy and manage.
+
+* Each additional dependency requires considerations in terms of monitoring, migrations, upgrades, durability among others.
+
+* ClickHouse is a relatively new product. Whereas tooling around PostgreSQL is very mature, we need to do some of the heavy lifting ourselves with ClickHouse.
+
+* Very few people know or have experience with managing a ClickHouse cluster relative to PostgreSQL. Our [partnerhship with Altinity](/blog/posthog-altinity-announce), which offers 24/7 technical support for ClickHouse, and bespoke training, is another way we're addressing this challenge. 
+
+* Simple operations such as application deployments or database upgrades have become complex automations
+
+To address this complexity and make PostHog easier to deploy, maintain and use for anyone, we gave ourselves two goals: **improve our test framework** and **improve our built-in monitoring**.
+
+This post is all about how we approached these goals, and our long-term roadmap for improving PostHog deployments.
```

Comment:
I'd cut this line tbh. 60 lines in, the reader should know what the post is about and standalone the meaning of this line is redundant.